Web/Text/src/WebARText.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { createElement, useEffect, useState } from "react";
+import { WebARTextContainerProps } from "../typings/WebARTextProps";
+import { MeshComponent, setAttributes } from "../../../Shared/ComponentParent/src/MeshComponent";
+import { useGizmoComponent } from "../../../Shared/ComponentParent/src/useGizmoComponent";
+import { Mesh, MeshBuilder, Scene, Texture, Vector3 } from "@babylonjs/core";
+import font from "../../../Shared/ComponentParent/utils/Open Sans_Regular.json";
+import "../../../Shared/ComponentParent/utils/injectEarcut";
+
+export function WebARText(props: WebARTextContainerProps): React.ReactElement {
+    const { mxMaterialTexture, mxText, name, tabIndex, style, ...restOfProps } = props;
+    const [mesh, setMesh] = useState<Mesh>();
```

Comment:
there's some duplication here with native ARText.
could this be combine into shared custom hooks?
especially because they uses same 3 useState: mesh, scene, and texture.

something like:
```
function useARText(){
   const [mesh, setMesh] = useState<Mesh>();
    const [scene, setScene] = useState<Scene>();
    const [texture, setTexture] = useState<Texture>();
    
    const gizmoTransform = useGizmoComponent({
        mesh: mesh,
        draggingEnabled: props.mxDraggingEnabled.value ?? false,
        pinchEnabled: props.mxPinchEnabled.value ?? false,
        rotationEnabled: props.mxPinchRotationEnabled.value ?? false,
        gizmoSize: Number(props.mxGizmoSize.value) ?? 0.1,
        color: props.mxGizmoColor.value ?? "#ffffff"
    });
    
    const handleSceneLoaded = (scene: Scene) => {
        const newMesh = MeshBuilder.CreateText("myText", mxText.value ?? "", font, {
            resolution: 64,
            depth: 10
        });
        if (newMesh) {
            newMesh.scaling = Vector3.Zero();
            setMesh(newMesh);
        }
        setScene(scene);
    };
    
    return [{data: {mesh, scene, texture}}, dispatch: {setMesh, setScene, setTexture}]
}
```

note: code probably not working, but you get the idea.